pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -188,6 +187,17 @@ pub struct DeltaLayer {
     inner: RwLock<DeltaLayerInner>,
 }
 
+impl std::fmt::Debug for DeltaLayer {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("DeltaLayer")
+            .field("key_range", &self.key_range)
+            .field("lsn_range", &self.lsn_range)
+            .field("file_size", &self.file_size)
+            .field("inner", &self.inner)
```

Comment:
I did not check, but at least `std::sync::Mutex` prints out the contents of a mutex if try_lock succeeds. Could be it's not safe/possible for `RwLock`, if so, this might not add any new information except for the type.

Reply:
It does `try_read()` so this is great: https://doc.rust-lang.org/stable/src/std/sync/rwlock.rs.html#482